moveos/moveos-stdlib/moveos-stdlib/sources/events.move

```diff
@@ -0,0 +1,23 @@
+/// events emit events to the event store.
+module moveos_std::events {
+    use moveos_std::tx_context::{Self, TxContext};
+    use moveos_std::object_id::ObjectID;
+
+    /// Emit a custom Move event, sending the data offchain.
+    ///
+    /// Used for creating custom indexes and tracking onchain
+    /// activity in a way that suits a specific application the most.
+    ///
+    /// The type T is the main way to index the event, and can contain
+    /// phantom parameters, eg emit(MyEvent<phantom T>).
+    public fun emit_event<T: drop + store>(ctx: &mut TxContext, event: T) {
+        let guid = tx_context::fresh_object_id(ctx);
```

Comment:
The original GUID is an event stream id representing the Event Type. the GUID + event sequence_number = EventID(or EventKey).

If we generate a Guid for everyone event, EventKey must also be changed.